app/src/main/java/com/jalasoft/routesapp/ui/routes/view/RouteSelectedFragment.kt

```diff
@@ -0,0 +1,204 @@
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+import com.google.android.gms.maps.model.*
+import com.jalasoft.routesapp.R
+import com.jalasoft.routesapp.data.model.remote.LinePath
+import com.jalasoft.routesapp.databinding.FragmentRouteSelectedBinding
+import com.jalasoft.routesapp.util.helpers.Constants
+import com.jalasoft.routesapp.util.helpers.GoogleMapsHelper
+
+class RouteSelectedFragment : Fragment(), OnMapReadyCallback {
+    private var _binding: FragmentRouteSelectedBinding? = null
+    private val binding get() = _binding!!
+    private lateinit var mFusedLocationClient: FusedLocationProviderClient
+    private var mMap: GoogleMap? = null
+    private var route: LinePath? = null
+    private var positionSelected = 0
+    private val requestFINELOCATIONPermissionLauncher =
+        registerForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted ->
+            checkPermissions(isGranted)
+        }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = FragmentRouteSelectedBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        route = arguments?.getSerializable(Constants.BUNDLE_KEY_ROUTE_SELECTED_DATA) as LinePath
+        positionSelected = arguments?.getInt(Constants.BUNDLE_KEY_ROUTE_SELECTED_POSITION) ?: 0
+
+        mFusedLocationClient = LocationServices.getFusedLocationProviderClient(requireContext())
+        val mapFragment = childFragmentManager.findFragmentById(R.id.map_route_fragment) as? SupportMapFragment
+        mapFragment?.getMapAsync(this)
+        btnActions()
+    }
+
+    override fun onMapReady(googleMap: GoogleMap) {
+        googleMap.setMapStyle(
+            MapStyleOptions.loadRawResourceStyle(
+                requireContext(),
+                R.raw.style_silver
+            )
+        )
+        mMap = googleMap
+        if (isLocationPermissionGranted()) {
+            getLocation()
+        }
+
+        val start = GoogleMapsHelper.locationToLatLng(route?.start!!)
+        val end = GoogleMapsHelper.locationToLatLng(route?.end!!)
+        mMap!!.addMarker(MarkerOptions().position(start).title(R.string.start_of_route.toString()).icon(GoogleMapsHelper.bitmapFromVector(requireContext(), R.drawable.ic_start_route)))
+        mMap!!.addMarker(MarkerOptions().position(end).title(R.string.end_of_route.toString()).icon(GoogleMapsHelper.bitmapFromVector(requireContext(), R.drawable.ic_end_route)))
+
+        addStopMarkers(route!!.stops)
```

Comment:
never force unwrap variables, validate if this route and nmap are different than null first